### PR TITLE
Formalise new editorial conventions for NEWS.md

### PR DESCRIPTION
**Pull request by hlandau**
This PR suggests a new editorial convention for NEWS.md. This is the convention already followed by #22785.

Summary of changes:

- The ordering and wording of the feature list has been restructured to better match our announcement blogposts for 3.2
- The structure has been broken down into sections with the most important news (potentially incompatible changes) at the top, followed by features, followed by documentation enhancements, followed by significant bugfixes (this section not currently present as we don't have any listed)
- A uniform wording is used for the feature list where suitable (not a hard rule). When followed, the first line is a summary line and has no trailing full stop. If more elaboration on a feature is needed this is added as a paragraph below in the same list item with a trailing full stop.
- RFCs are always cited with a space (RFC 9000)
- Sort features in descending order of significance
- Links to starting points and community resources

See dev/NEWS-FORMAT.md for the details. Feedback welcome.

I'd also consider changing `### Major changes between OpenSSL 3.2.0 and 3.2.1 [1 Jan 2023]` to `### Release notes for OpenSSL 3.2.0 [1 Jan 2023]`. Thoughts?

### CODE REVIEW

**Comment by hlandau** [comment]
Changed this to be the editorial conventions document only now that the release notes for 3.2.0 are in another PR.

File: dev/NEWS-FORMAT.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+```
+
+The editorial conventions for this block are similar to those for feature
+additions, except that the verb is part of the summary line. The other rules are
+the same.
+
+For consistency, use the wording `Added ...` or `Clarified ...` as the summary
+line if possible.
+
+#### Listing bug fixes and miitgations
```
Comment (levitte) [inline dev/NEWS-FORMAT.md:175]: Nit: spelling

Side note: I was curious and went looking if MIIT means something.  Ah yup, it's apparently China's Ministry of Industry and Information Technology

**Comment by hlandau** [comment]
I think we should consider having the release notes available as a page on our website also (as HTML). We can probably automatically generate this.

**Comment by hlandau** [comment]
Fixed nit

**Comment by levitte** [comment]
I've pondered going further; other projects I've seen have a per release line Release Notes file.  For us, that would mean separate files for 3.0, 3.1, ...  We could even go so far as to making the branch specific.  That would help publishing quite a lot.

**Comment by levitte** [comment]
For old NEWS, copy that to another file, like OLD-NEWS.md?

**Review by mattcaswell — APPROVED** [review]

**Review by levitte — APPROVED** [review]
I'm pre-approving cherry-picking this to `openssl-3.2` as well.

**Comment by hlandau** [comment]
@mattcaswell Approve for 3.2?

**Comment by mattcaswell** [comment]
Yes - ok for 3.2

**Comment by hlandau** [comment]
Merged to master and 3.2. Thanks for the reviews.

**Pull request closed without merging**